### ISSUE TITLE
Messages from sender not already in chat list do not display.

### ISSUE THREAD

**Issue by jelbana** [snikket-im/snikket-ios#161]
Unless the app is open at the time of receiving a message, a new chat wont display in the chat list. Push notifications indicate a message was delivered, but opening the app fails to display the message and the new chat. If a subsequent message is sent while the app is open, then the new chat is added to the chat list and previous messages are visible. 

Snikket Version 1.2 build 8
iOS 15.1

**Comment by jelbana** [comment]
Just wanted to add that this is a pretty serious issue. I'm effectively unable to use the app since new chats do not appear in the chat list if the app is in the background.

**Comment by mwild1** [comment]
I've attempted to reproduce this issue on multiple devices and multiple accounts. New messages are always showing up at the top of the list.

If anyone can confirm they are seeing this issue in the latest version of the app, and/or know under what specific circumstances this behaviour occurs, that would be very helpful.

**Comment by glward** [comment]
See my comment on #194 -- I'm able to consistently reproduce this by getting an SMS message from someone not in my roster via jmp.chat (e.g from a delivery service or similar).

Server is the latest Snikket self-hosted container (beta/9916f76a8f96), client is Snikket iOS 1.2 build 10 on both iOS (iPhone 13 mini) and iPad OS (iPad mini, iPad Pro 11).

Typically I'll get a push notification but when I open the app no new chat is open, and the notification badge stays stuck until I clear the notification from notification center. Trying to open the app from the notification doesn't help -- the chat still never opens.

Usually I can open a chat with the +...@cheogram.com after adding it to the roster and the history will load correctly at that point.

**Comment by mwild1** [comment]
This is exactly what I've been doing... but the message always shows in the list as soon as the app opens. I've tried with the app open, the app closed, the device locked, and so on.

*However,* since my previous comment I think I found a way to reproduce it (or something similar). If I disable the network (e.g. wifi) while the message is sent to the iOS device, and then enable the network again and open the app, the app is not showing the message. I'm going to work on debugging that, and hopefully it's the same issue!

Also, thanks for linking to #194. I thought I remembered more discussion on this problem than I saw here.
